### DISCUSSION TITLE
Unregister a field which is a dep for another field

### DISCUSSION THREAD

**Discussion by ktalalykin** [react-hook-form/react-hook-form#8651] [Q&A]
Hi! I have two inputs where validation depends on each other through `deps`. Now, imagine duplicate validation of values in those two inputs, hence if two values are the same, both fields would have error. Now, i delete and unregister one of the fields, error doesnt disappear automatically, it persists until i touch the field. Anything that I can do about this.

**Comment by bluebill1049** [comment]
can you use `watch` for that? and plus `shouldUnregister` set to `true`?

**Reply by ktalalykin** [reply]
im not sure watch can help because with `deps` usage watch will add additional calls to validate the form if i understood correctly what you meant. I came up with a solution, where for my generic input on unmount i would call `trigger` on its deps (sandbox https://codesandbox.io/s/usecontroller-forked-qjxw5x?file=/src/App.js), but im not sure if its best solution and if im not fully utilising RHF.

**Comment by ktalalykin** [comment]
Tbh, my solution doesnt work. More in depth explanation, i have field array of inputs, where the values should be codependent to evaluate the uniqueness. Codependency i can implement through deps, listing all the fields names from field array , however, i need a delete functionality, where if there are two inputs with non unique values showing error, deleting one of the inputs will remove the error for the other one, but i can get it working.
